pkg/tangy/rpm.go

```diff
@@ -0,0 +1,51 @@
+package tangy
+
+import (
+	"context"
+	"strings"
+
+	"github.com/jackc/pgx/v5"
+)
+
+type RpmPackageSearch struct {
+	Name    string
+	Summary string
+}
+
+func (t *tangyImpl) RpmRepositoryVersionPackageSearch(ctx context.Context, hrefs []string, search string) ([]RpmPackageSearch, error) {
+	conn, err := t.pool.Acquire(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Release()
+
+	query := `
+		SELECT DISTINCT ON (rp.name) rp.name, rp.summary
+		FROM core_repositoryversion crv
+		INNER JOIN core_repository cr on crv.repository_id = cr.pulp_id
+		INNER JOIN core_repositorycontent crc on crv.pulp_id = crc.version_added_id
```

Comment:
would version_added cause this to only be returned for the version that actually added the RPM?

Reply:
i think it would, so I changed it to

```
SELECT DISTINCT ON (rp.name) rp.name, rp.summary
		FROM core_repositoryversion crv
		INNER JOIN core_repository cr on crv.repository_id = cr.pulp_id
		INNER JOIN core_repositorycontent crc on cr.pulp_id = crc.repository_id
		INNER JOIN core_content cc on crc.content_id = cc.pulp_id
		INNER JOIN rpm_package rp on cc.pulp_id = rp.content_ptr_id
		WHERE CONCAT(crv.repository_id, '/', crv.number) = ANY ($1)
		AND rp.name ILIKE CONCAT( '%', $2::text, '%')
```

Reply:
instead of trying to test manually, i updated your test to simulate re-syncing a repository to produce two different versions:  https://gist.github.com/jlsherrill/5008aaee911029e74edb22dc014cd245

It all worked except when i search the first version after having created the second version, i seem to be getting the second version's results?